terrain-manager.js

```diff
@@ -0,0 +1,180 @@
+import * as THREE from 'three';
+
+export class TerrainManager {
+
+	constructor(chunkSize, range, geometryUtils) {
+
+		this.chunkRange = range;
+		this.chunkSize = chunkSize;
+		this.geometryUtils = geometryUtils;
+
+		this.center = new THREE.Vector3();
+		this.chunkCount = this.chunkRange * 2 + 1;
+
+		this.targetChunkIds = this._calculateTargetChunks();
+		this.currentChunks = this.targetChunkIds.map((v, i) => { return { slots: [i, i], chunkId: v } });
+
+		this.segment = 16;
+
+		/*
+		 * if following parameters are too small, memory areas of chunks can be overlaid
+		 * if too big, memory will be over allocated;
+		 */
+		this.vertexBufferSizeParam = 20;
+		this.indexBufferSizeParam = 20;
+
+		this.init();
+	}
+
+	init() {
+
+		this._generateBuffers();
+	}
+
+	_generateBuffers() {
+
+		this.bufferFactory = this.geometryUtils.generateTerrain(
```

Comment:
This is not a good name. Is this generating all terrain for all chunks? It runs on init so that seems impossible..

Alternatively, is it allocating memory for future generation? If so, then why is it called `generateTerrain` when it does not generate terrain?

Reply:
It allocates the memory and generates initial chunks